### ISSUE TITLE
Add a `low rank` update type

### ISSUE THREAD

**Issue by blester125** [r-three/git-theta#108]
To start with, we are going to have the `low rank` update type re-compute the low-rank update based on the current and previous value.

If the user includes the low-rank parameters as part of the computation graph, we will just save and track those as normal. The main parameter will most likely not change much. Nothing special happens in this case.

Eventually we may want to have the user supply the update values to avoid the work of them applying the update to the parameter and use unapplying it but that has UI difficulties.

**Comment by blester125** [comment]
I ran into an issue trying to infer the low rank update with SVD like we discussed in the meeting. We can run SVD on the update (the different of the previous and current parameters) we will get back the `min(IN, OUT)` singular values which is generally > K, our low rank update value.

Many of these `s` values are very close to zero so we can take only the top K values (ignoring how we will infer/be supplied K for now) and recreate an update that passes `np.allclose` when compared to the real update, however, it does not pass hash equality. Even if we use all `s` values (which would actually cost more to store than a full update for square matrices) we still don't get a perfect match or pass hash equality.

Demonstration of this behavior can be found [here]( https://gist.github.com/blester125/981f139e30b601b1756ed3c0376e86da)

np.allclose isn't transitive so we can't invent new hashing method that means all things that pass np.allclose will hash to the same value.

This mismatch will cause issues because the clean hash is computed based on the parameters on disk but when we smudge we will get slightly different parameters (created from the application of the close enough low rank update) causing a diff to be detected right after checkout (and a desync between what hash is record and what hash is true).

It seems like we need a way for a user to specify low-rank update values that we can load and use directly, or some sort of overhaul of how we record hashes/test for equality (i.e. if we don't hash equal should we fall back to `np.allclose` to a "true" equality check? Does that mean we would suppress reporting of the diff in our diff tool? etc.)

@nkandpa2 any thoughts? or is there a better way to infer the low rank updates that is more likely to match? I also tried running LBFGS optimization in pytorch (loss was MSE between `A @ B` and the true update) and that has similar results, `np.allclose` passed but hashing did not.
